app/src/main/java/org/p2p/wallet/auth/interactor/MetadataInteractor.kt

```diff
@@ -90,8 +90,8 @@ class MetadataInteractor(
     private fun saveMetadataToStorage(metadata: GatewayOnboardingMetadata?) {
         // TODO PWN-8771 - implement database for metadata
         val ethAddress = getEthereumPublicKey().orEmpty()
-        secureStorageContract.saveObject(
-            key = SecureStorageContract.Key.KEY_ONBOARDING_METADATA.withCustomKey(ethAddress),
+        accountStorage.saveObject(
```

Comment:
sorry, but no. 
You need a migration code for all of this.
Extract everything to MetadataProvider and rule there